include/matrix_assembler.cc

```diff
@@ -0,0 +1,151 @@
+// Constructor
+MatrixAssembler::MatrixAssembler() {
+  jac_ = std::make_shared<Eigen::SparseMatrix<double>>();
+  node_head_vec_ = std::make_shared<Eigen::VectorXd>();
+  node_discharge_vec_ = std::make_shared<Eigen::VectorXd>();
+  pipe_discharge_vec_ = std::make_shared<Eigen::VectorXd>();
+};
+
+// Assign global index to nodes in the whole network
+void MatrixAssembler::assign_global_nodal_indices(
```

Comment:
If the mesh class has all the nodes, then do we need this global index? Can we not use the index from the mesh class itself?

Reply:
Even if we have several meshes it is easier to just pass the start of the global index to the mesh class and use that as a starting point. We don't need  separate global index.